src/core/Tags.java

```diff
@@ -637,4 +637,19 @@ public static boolean looksLikeInteger(final String value) {
     return true;
   }
 
+  /**
+   * Returns true if the given string can fit into a float.
+   * @param value The String holding the float value.
+   * @return true if the value can fit into a float, false otherwise.
+   * @throws NumberFormatException if the value is not numeric.
+   * @since 2.0.2
+   */
+  public static boolean fitsInFloat(final String value) {
+    final float f = Float.parseFloat(value);
+    final String converted = Float.toString(f);
+
+    // this will be false if there was a loss of precision.
+    return value.equals(converted);
```

Comment:
I don't think this is the proper way to determine if the precision is lost as some floating point values won't survive the change due to the fact that some values can't be represented in base 2. Instead I saw one way was to parse to Double, then cast to float and see if the result was infinite or not. But that didn't seem to help with negative floats so we'd have to figure that one out.

http://docs.oracle.com/cd/E19957-01/806-3568/ncg_goldberg.html